docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -83,15 +83,15 @@
    private static final String NO_IMAGE = "jclouds-placeholder-for-image";
 
    protected List<String> dns = ImmutableList.of();
-   protected String hostname;
-   protected Integer memory;
-   protected Integer cpuShares;
-   protected List<String> entrypoint = ImmutableList.of();
-   protected List<String> commands = ImmutableList.of();
+   @Nullable protected String hostname;
+   @Nullable protected Integer memory;
+   @Nullable protected Integer cpuShares;
+   @Nullable List<String> entrypoint;
+   @Nullable List<String> commands;
```

Comment:
FYI, the JSON that the Docker API recieves will treat null differently to an empty list. So for the entrypoint value, null means 'do nothing, just use the values from the image' but an empty list means 'change this value to an empty list, and discard the value from the image' which is why we need to be `@Nullable` here.